sys/net/gnrc/netif/ieee802154/gnrc_netif_ieee802154.c

```diff
@@ -268,14 +297,43 @@ static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
         DEBUG("_send_ieee802154: Error preperaring frame\n");
         return -EINVAL;
     }
-
-    /* prepare iolist for netdev / mac layer */
-    iolist_t iolist = {
-        .iol_next = (iolist_t *)pkt->next,
+    iolist_t iolist_payload = {
+        .iol_next = (iolist_t *)pkt->next->next,
+        .iol_base = pkt->next->data,
+        .iol_len = pkt->next->size
+    };
+    iolist_t iolist_header = {
+        .iol_next = &iolist_payload,
         .iol_base = mhr,
-        .iol_len = (size_t)res
+        .iol_len = res
     };
-
+#if IS_USED(MODULE_IEEE802154_SECURITY)
+    gnrc_pktbuf_merge(pkt->next);
```

Comment:
Due tot his write action, `pkt->next` needs to be write-protected first (see [doc of that function](https://doc.riot-os.org/group__net__gnrc__pktbuf.html#gae41835e987f693706a388a3d9e14f83a). To my knowledge there was no write protection issued granted before in the send operation, as it usually just copies data over from the snip (and releases of course, but those do not need to be write-protected). Also, the function should be checked for errors and return as such.

```suggestion
    /* write protect `pkt` to set `pkt->next` */
    gnrc_pktsnip_t *tmp = gnrc_pktbuf_start_write(pkt);
    if (tmp == NULL) {
        DEBUG("...");
        return -ENOMEM;
    }
    pkt = tmp;
    tmp = gnrc_pktbuf_start_write(pkt->next);
    if (tmp == NULL) {
        DEBUG("...");
        return -ENOMEM;
    }
    pkt->next = tmp->next;
    res = gnrc_pktbuf_merge(pkt->next);
    if (res < 0) {
        return res;
    }
```

I am sure there is a nicer way to code this, but I wanted to get this out quickly before this PR is merged.